### DISCUSSION TITLE
Replacement of .IsSameOrEqualTo() for extensions in v6.0

### DISCUSSION THREAD

**Discussion by CNBoland** [fluentassertions/fluentassertions#1644] [Q&A] [answered]
I'm working with an extension that uses `IsSameOrEqualTo()` like so:
```cs
public TObjectResultAssertion WithValue(object expectedValue, string reason = "", params object[] reasonArgs)
{
    object actualValue = ObjectResultSubject.Value;

    Execute.Assertion
        .BecauseOf(reason, reasonArgs)
        .ForCondition(actualValue.IsSameOrEqualTo(expectedValue))
        .WithDefaultIdentifier(Identifier + ".Value")
        .FailWith(FailureMessages.CommonFailMessage, expectedValue, actualValue);

    return (TObjectResultAssertion)this;
}
```
`IsSameOrEqualTo()` is now an internal in v6.0. What is a recommended replacement?

**Comment by dennisdoomen** [comment, marked as answer]
Since it's an internal piece of code, it was never meant to be reused outside our code base. But you can still copy the code if you really need it. Although in most cases, a simple `Equals` should suffice.

**Reply by jnyrup** [reply]
To elaborate a bit more, `IsSameOrEquals` is now mostly `object.Equals()` plus 
* allow comparison of boxed numerics, e.g. comparing `(object)1d` and `(object)1L`, and 
* optimizations to avoid boxing of value types.

**Reply by CNBoland** [reply]
@dennisdoomen That worked nicely. Thank you for the suggestion.

**Comment by Lemraj** [comment]
Hi,
I have upgraded to version 6.10.0 and just replaced `IsSameOrEqualTo()` with `Equals()` as suggested. But i am getting a `System.NullReferenceException`. what am i doing wrong?

```cs
        if (actualValue.Equals(expectedValue.Value))
        {
        }
````

**Reply by jnyrup** [reply]
With `object.Equals()` I guess I meant [`public static bool Equals(object? objA, object? objB)`](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Object.cs,53)

So in your case `Equals(actualValue, expectedValue.Value)`.

Here's the implementation of [`IsSameOrEqualTo`](https://github.com/fluentassertions/fluentassertions/blob/5.10.3/Src/FluentAssertions/Common/ObjectExtensions.cs) from v5

**Reply by Lemraj** [reply]
Thanks for the quick response! your suggestion worked for me.
